src/cms/data/functions/loadOperationData.js

```diff
@@ -0,0 +1,220 @@
+        try {
+          // get existing record
+          let existingRecord;
+          switch (modelName) {
+            case "park-operation":
+              // shoud only be one record per protected area
+              existingRecord = currentData.find(d => (d.protectedArea) && (d.protectedArea.orcs === data.orcs))
+              break;
+            case "park-operation-sub-area-date":
+              // specific call to get existingRecord
+              // both subarea and year
+              existingRecord = currentData.find(d => (d.parkOperationSubArea?.parkSubAreaId === data.parkSubAreaId) && (d.operatingYear === data.operatingYear)); // get existing record
+              break;
+            default:
+              existingRecord = currentData.find(d => d[idName] === idVal);
+              break;
+          }
+          const doesExist = (existingRecord !== undefined);
+
+          if (!idVal) {
+            // id needed
+            console.log("Id for " + modelName + " is missing in json: " + idName)
+            missingCount++;
+          } else {
+
+            const keys = Object.keys(data); // replace empty strings with nulls
+            for (let i = 0; i < keys.length; i++) {
+              if (data[keys[i]] === "") data[keys[i]] = null;
+            }
+            
+            let isMissingRelation = false;
+
+            switch (modelName) { // add relations, specific to model
+              case "park-operation-sub-area":
+                // make relation between sub-area and type
+                if (data.parkSubAreaTypeId) {
+                  const subAreaTypeService = strapi.services["park-operation-sub-area-type"];
+                  const subAreaType = await subAreaTypeService.findOne({ subAreaTypeId: data.parkSubAreaTypeId });
+                  data.parkSubAreaType = subAreaType ? subAreaType.id : null;
+                } else {
+                  data.parkSubAreaType = null;
+                }
+                // relate subarea to protected areas
+                if (data.orcs) {
+                  const protectedAreaService = strapi.services["protected-area"];
+                  const protectedArea = await protectedAreaService.findOne({ orcs: data.orcs })
+                  data.protectedArea = protectedArea ? protectedArea.id : null;
+                } else {
+                  data.protectedArea = null;
+                }
+                // relate subarea to facility
+                if (data.parkFacilityId) {
+                  const parkFacilityService = strapi.services["park-facility"];
+                  const parkFacility = await parkFacilityService.findOne({ id: data.parkFacilityId })
+                  data.parkFacility = parkFacility ? parkFacility.id : null;
+                } else {
+                  data.parkFacility = null;
+                }
+                isMissingRelation = ((data.parkSubAreaType === null) || (data.protectedArea === null)) // park facility not needed
+                break;
```

Comment:
Drop into a function call instead of inline `case:` (same below)

Reply:
refactored case statements